airdrop/SPECIFICATION.md

```diff
@@ -0,0 +1,233 @@
+Table of Contents
+- [Overview](#overview)
+- [Use Cases](#use-cases)
+  - [Actions that users are able to perform](#actions-that-users-are-able-to-perform)
+    - [Create Airdrop](#create-airdrop)
+    - [Claim Airdrop](#claim-airdrop)
+  - [Information that is presented to users](#information-that-is-presented-to-users)
+    - [Airdrops](#airdrops)
+    - [Balances](#balances)
+    - [Discovery of Airdrops](#discovery-of-airdrops)
+- [Specification](#specification)
+  - [Airdrop-Distributor Smart Contract](#airdrop-distributor-smart-contract)
+    - [Starting an Airdrop](#starting-an-airdrop)
+      - [Context](#context)
+      - [Requirements](#requirements)
+    - [Claiming an Airdrop](#claiming-an-airdrop)
+      - [Context](#context-1)
+      - [Requirements](#requirements-1)
+    - [Helper functions for the user interface](#helper-functions-for-the-user-interface)
+      - [End Block](#end-block)
+        - [Context](#context-2)
+        - [Requirements](#requirements-2)
+      - [Merkle Root](#merkle-root)
+        - [Context](#context-3)
+        - [Requirements](#requirements-3)
+  - [Simple-Token Smart Contract](#simple-token-smart-contract)
+    - [Creating A Token](#creating-a-token)
+      - [Context](#context-4)
+      - [Requirements](#requirements-4)
+    - [Minting Tokens](#minting-tokens)
+      - [Context](#context-5)
+      - [Requirements](#requirements-5)
+  - [User interface](#user-interface)
+
+# Overview
+
+This document provides an overview of the application.
+
+It outlines the use cases, i.e. desirable functionality, in addition to requirements for the smart contract and the user interface.
+
+This application inherits specification from the [Binary Merkle Proof Verification Library specification](https://github.com/FuelLabs/sway-libs/blob/master/sway_libs/src/merkle_proof/SPECIFICATION.md).
+
+# Use Cases
+
+This section contains general information about the functionality of the application and thus does not touch upon any technical aspects.
+
+If you are interested in a functional overview then this is the section for you.
+
+## Actions that users are able to perform
+
+This sub-section details what a user is able to do e.g. click a button and "x, y, z" happens.
+
+### Create Airdrop
+
+A user should be able to create an airdrop which consists of 
+
+1. A token which will be distributed
+2. A Merkle root constructed from a Merkle Tree where each leaf is the resulting `sha-256` hash of a `(user address, amount)` tuple
+3. A length of time in which a claim is valid
+
+### Claim Airdrop
+
+A user should be able to claim their airdrop
+
+1. Before the claim period is over
+2. If they has not yet claimed
+3. If the amount they are claiming is verifiable

```

Comment:
I've added this suggestion however, excluded the "within". This would suggest the user can claim a smaller portion of tokens than allottedwhich is not true. They are only able to claim the exact amount of tokens they have been allotted.